### PR TITLE
test(config): add unit tests to config flags

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

the initialization of `config.flags` is not tested

GitHub Issue Number: N/A

## What is the new behavior?

this commit adds unit tests to the portion of the configuration
validation process for the 'flags' field

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

unit tests pass

### CODE REVIEW

File: src/compiler/config/test/validate-config.spec.ts [hunk 1]
```diff
@@ -18,6 +18,48 @@ describe('validation', () => {
...
+        expect(config.devMode).toBe(true);
+      });
+
+      it('defaults "devMode" to false when "flag.prod" & "flag.dev" are truthy', () => {
+        userConfig.flags = { dev: true, prod: true };
+        const { config } = validateConfig(userConfig);
+        expect(config.devMode).toBe(false);
+      });
+
+      it('sets "devMode" to false the user provided flag isn\'t a boolean', () => {
```
Comment (alicewriteswrongs) [inline src/compiler/config/test/validate-config.spec.ts:53]:
```suggestion
      it('sets "devMode" to false if the user provided flag isn\'t a boolean', () => {
```

Comment (alicewriteswrongs) [reply src/compiler/config/test/validate-config.spec.ts:53]: or `"when"`, possibly?

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
just one thing I noticed

File: src/compiler/config/test/validate-config.spec.ts (hunk 1 above)
Comment (rwaskiewicz) [reply src/compiler/config/test/validate-config.spec.ts:53]: Yeah, that makes sense!

**Review by alicewriteswrongs — APPROVED** [review]
lgtm!

**Pull request merged**